src/java/org/apache/cassandra/index/sai/disk/v1/V1SearchableIndex.java

```diff
@@ -169,6 +175,12 @@ public List<RangeIterator<Long>> search(Expression expression,
     {
         List<RangeIterator<Long>> iterators = new ArrayList<>();
 
+        if(keyRange instanceof Bounds && keyRange.left.equals(keyRange.right) && keyRange.left instanceof DecoratedKey)
+        {
+            if (!filter.isPresent((DecoratedKey)keyRange.left))
```

Comment:
```suggestion
            if (!sstableContext.sstable().getBloomFilter().isPresent((DecoratedKey)keyRange.left))
```

It'd be better to fetch BF per request instead of using it as instance variable because BF is loaded lazily based on sstable hotness in CNDB. Converged Cassandra hasn't ported relevant CNDB changes yet.